@Vedant794 this is a known issue, it's on our backlog to fix that.
But the previous step also fail with compilation issue:

```
/home/runner/work/kestra/kestra/core/src/test/java/io/kestra/plugin/core/namespace/DownloadFilesTest.java:74: error: cannot find symbol
        assertThat(output.getFiles()).containsKey(weirdFileName);
                                                  ^
  symbol:   variable weirdFileName
  location: class DownloadFilesTest
/home/runner/work/kestra/kestra/core/src/test/java/io/kestra/plugin/core/namespace/DownloadFilesTest.java:75: error: cannot find symbol
        assertThat(output.getFiles().get(weirdFileName)).isNotNull();
                                         ^
  symbol:   variable weirdFileName
  location: class DownloadFilesTest
```

Please take care at least that the project is building using `gradle compileTestJava` and the test is passing by for ex running it inside your IDE or using `gradle test`.

--- a/core/src/test/java/io/kestra/plugin/core/namespace/DownloadFilesTest.java
+++ b/core/src/test/java/io/kestra/plugin/core/namespace/DownloadFilesTest.java
@@ -71,7 +71,7 @@ public class DownloadFilesTest {
         // Run the task
         DownloadFiles.Output output = downloadFiles.run(runContext);
         // Verify the file was downloaded successfully
-        assertThat(output.getFiles()).containsKey(weirdFileName);
-        assertThat(output.getFiles().get(weirdFileName)).isNotNull();
+        assertThat(output.getFiles()).containsKey("/a/b/sample.file with weird naming.txt");
+        assertThat(output.getFiles().get("/a/b/sample.file with weird naming.txt")).isNotNull();
     }
 }

